src/common/util/parse-service-data.ts

```diff
@@ -0,0 +1,7 @@
+export const parseServiceData = (serviceData: string): object => {
```

Comment:
This has nothing to do with service data. I would probably call it something like safeJsonParse.

```ts
export const safeJsonParse <ReturnValue>(value: string, default: ReturnValue) {
  try {
    return value ? JSON.parse(value) as Returnvalue : default
  } catch (err) {
    return default;
  }
}
```

However, this is very dangerous, because now we're ignoring invalid JSON and just throwing it away. I doubt this is a valid usecase for anything, imagine your service data not being used.